### PR TITLE
[Synthetics] Allow specifying PriorityClass for pods

### PR DESCRIPTION
**Pull request by tr3mor**
#### What this PR does / why we need it:

This PR allows user to specify [PriorityClass](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) they want to use for Private Location PODs. 
My use case is to guarantee that Private Location pods have higher priority than application pods during scheduling and can't be evicted by app pods in case of resource pressure on the k8s node.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`
- [ ] For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)

### CODE REVIEW

**Review by loutPhilipps — APPROVED** [review]
LGTM

**Comment by tr3mor** [comment]
Thank you for reviewing @loutPhilipps! What is the policy for merging? I want to deploy using this chart for one of our locations.

**Review by loutPhilipps — APPROVED** [review]

**Comment by loutPhilipps** [comment]
> Thank you for reviewing @loutPhilipps! What is the policy for merging? I want to deploy using this chart for one of our locations.

I can merge this as soon as the CI is all green

**Comment by loutPhilipps** [comment]
> > Thank you for reviewing @loutPhilipps! What is the policy for merging? I want to deploy using this chart for one of our locations.
> 
> I can merge this as soon as the CI is all green

Actually @tr3mor it looks like I cannot merge this PR since your commits need to be signed, could you please rewrite them and sign them? After that is done we should be all good to merge this

**Comment by tr3mor** [comment]
It would be great to have it mentioned in CONTRIBUTION.md. Anyway, done, squashed, signed, and force-pushed @loutPhilipps

**Comment by loutPhilipps** [comment]
> It would be great to have it mentioned in CONTRIBUTION.md

Thanks for reporting that, it was now been added in https://github.com/DataDog/helm-charts/pull/1287

**Comment by loutPhilipps** [comment]
@tr3mor you should be able to use https://github.com/DataDog/helm-charts/releases/tag/synthetics-private-location-0.15.23 now. Thanks for your contribution and have a nice day!

**Pull request merged**
